src/libmongoc/tests/unified/operation.c

```diff
@@ -242,19 +265,34 @@ operation_list_collections (test_t *test,
    bool ret = false;
    mongoc_database_t *db = NULL;
    mongoc_cursor_t *cursor = NULL;
+   bson_parser_t *parser = NULL;
```

Comment:
I noticed this PR is still missing some operations. For instance, `operation_drop_collection` was not modified. This is likely because a variant of `mongoc_collection_drop` was never introduced to take a "sessionId" option. Since you're going through the entire file here, would you mind creating a list of all of the operations that still _do not_ take a session operation (excluding things like the internal assertion and special test operations) and dump that list into a CDRIVER ticket? I think that would provide a good starting point for any future plans to create the necessary APIs so that all session-supporting operations _can_ solicit a "sessionId" option.

Reply:
So I switched to using `mongoc_collection_drop_with_opts` which _does_ [take a session ID](http://mongoc.org/libmongoc/current/mongoc_collection_drop_with_opts.html). After changing that, and adding sessionId to all the `find_and_modify` operations, I think we're pretty close to everything. I don't think there are any operations represented here that are session-supporting but _cannot_ solicit a "sessionId" option.

The few that I'm wondering about are: `operation_failpoint`, `assert_index_exists`, and `assert_collection_exists`. I _can_ append a sessionId to all of these since they use `mongoc_client_command_simple`, [mongoc_collection_find_indexes_with_opts](http://mongoc.org/libmongoc/current/mongoc_collection_find_indexes_with_opts.html) and [mongoc_database_get_collection_names_with_opts](http://mongoc.org/libmongoc/current/mongoc_collection_find_indexes_with_opts.html) respectively.

However, I don't see any usages of `session` with those three operations in existing unified format tests. Thoughts?

Reply:
I don't see `mongoc_collection_drop_with_opts` in this PR. Was that not pushed?

If a [special test operation](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#special-test-operations) doesn't document a session option, those should not support a session at all.

Reply:
Ah 🤦 forgot to push. Should be all set now.

And great thanks for linking me to the spec there; it looks like none of `failPoint`, `assertIndex(Not)Exists` or `assertCollection(Not)Exists` are documented as taking a session option, so I will not add sessionId to those operations.